### PR TITLE
Add regex to star

### PR DESCRIPTION
**Pull request by robertdefilippi**
This is a:
- [ ] bug fix PR with no breaking changes — please ensure the base branch is `master`
- [x] new functionality — please ensure the base branch is the latest `dev/` branch
- [ ] a breaking change — please ensure the base branch is the latest `dev/` branch

## Description & motivation
Adding regex functionality to `dbt_utils.star`

## Checklist
- [x] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [x] BigQuery
    - [x] Postgres
    - [x] Redshift
    - [x] Snowflake
- [ ] I followed guidelines to ensure that my changes will work on "non-core" adapters by:
    - [ ] dispatching any new macro(s) so non-core adapters can also use them (e.g. [the `star()` source](https://github.com/fishtown-analytics/dbt-utils/blob/master/macros/sql/star.sql))
    - [ ] using the `limit_zero()` macro in place of the literal string: `limit 0`
    - [ ] using `dbt_utils.type_*` macros instead of explicit datatypes (e.g. `dbt_utils.type_timestamp()` instead of `TIMESTAMP`
- [x] I have updated the README.md (if applicable)
- [x] I have added tests & descriptions to my models (and macros if applicable)
- [ ] I have added an entry to CHANGELOG.md

### CODE REVIEW

**Comment by robertdefilippi** [comment]
I'm having some issues testing this from the instructions in the `./integration_tests`.

When I try to `test` I get the following error:

```
# ./dbt-utils/integration_tests

$ make test target=postgres
docker-compose -f ../docker-compose.yml up dbt
[+] Running 2/2
 ⠿ Container dbt-utils_postgres_1  Created                                                                                                                                   0.0s
 ⠿ Container dbt-utils_dbt_1       Recreated                                                                                                                                 0.2s
Attaching to dbt_1
dbt_1  | Running with dbt=1.0.0-b2
dbt_1  | No profile "integration_tests" found, continuing with no target
dbt_1  | Installing ../
dbt_1  |   Installed from <local @ ../>
dbt_1  | Running with dbt=1.0.0-b2
dbt_1  | Encountered an error while reading profiles:
dbt_1  |   ERROR Runtime Error
dbt_1  |   Compilation Error
dbt_1  |     Could not render {{ env_var('POSTGRES_TEST_HOST') }}: Env var required but not provided: 'POSTGRES_TEST_HOST'
dbt_1  | Defined profiles:
dbt_1  |  - integration_tests
dbt_1  |
dbt_1  | For more information on configuring profiles, please consult the dbt docs:
...
```

How can I fix this so I can test my PR.

Thanks!

**Comment by robertdefilippi** [comment]
Hey @drewbanin, sorry to bother you about this. Would you have an answer to the above? I'm trying to get the CI jobs to run to test this PR, but I'm unsure how to do that.

**Comment by joellabes** [comment]
@robertdefilippi have a look at the https://github.com/dbt-labs/dbt-utils/blob/main/run_test.sh script which is triggered by Circle. 

It looks like the profile is populated by [copying](https://github.com/dbt-labs/dbt-utils/blob/main/run_test.sh#L17) from `ci/sample.profiles.yml`, which doesn't actually exist in this project anymore 😕 

I just tried to look at the webhook history for this, but GitHub only retains them for a month. 

Could you push something else up to this PR branch? I think that should trigger another webhook and then I can see what happens after that.

I'm broadly on board with bringing this in though, once we get the tests working 👍

**Comment by robertdefilippi** [comment]
@joellabes Just pushed a changed. Would love to figure out what the issues is 😄

**Comment by robertdefilippi** [comment]
Not a problem @joellabes ! This has been sitting for a month, so a little longer won't be an issue. Thank you for looking into this 😃

**Comment by joellabes** [comment]
OK I heard back from them! Sounds like if you try again it'll work? 🤞 
 
> When checking our logs for the X-GitHub-Delivery id XXXXXX, I saw that we were getting 401 errors for the user robertdefilippi from Github.
> 
> This can happen when a user has a CircleCI account, but hasn't logged in to CircleCI in a long time, and we have a stale token cached for them on our end. When this happens, builds may be blocked, and you will see a 400 webhook error on Github.
> 
> For PRs created/updated by users who do not have a CircleCI account, this issue should not block your builds.
> 
> I have refreshed the cache for the user `robertdefilippi, but could you have them try to push another commit to the branch that has an open PR?

**Comment by robertdefilippi** [comment]
@joellabes finally got this all figured out 🙂  Thanks for all your help through this!

**Comment by joellabes** [comment]
Hurrah! I'll check it out start of next week 🔜

**Comment by joellabes** [comment]
@robertdefilippi sorry for the slow moving here! My current plan is to ship [0.8.1 tomorrow-ish](https://github.com/dbt-labs/dbt-utils/pull/502), and then include your changes in 0.9.0 of utils sometime after that (because the change of method signature will be a breaking change for anyone using positional arguments), which will be a wee way away because it will also be where we pull some stuff out of utils as discussed in #487 

Fair warning, 0.8.1 includes a fair bit of movement around the star macro - shouldn't be insurmountable but I think you'll find some merge conflicts sorry. It'd be worth changing the target branch to `next/minor` once 0.8.1 is out.

**Comment by robertdefilippi** [comment]
@joellabes got it! Thanks for all your help so far, and the heads up with the potential changes coming my way 😄 

Once 8.1 is out I'll have a go and fixing the conflicts.

File: macros/sql/star.sql [hunk 1]
```diff
@@ -1,8 +1,8 @@
-{% macro star(from, relation_alias=False, except=[], prefix='', suffix='') -%}
-    {{ return(adapter.dispatch('star', 'dbt_utils')(from, relation_alias, except, prefix, suffix)) }}
+{% macro star(from, relation_alias=False, except=[], regex='', prefix='', suffix='') -%}
+    {{ return(adapter.dispatch('star', 'dbt_utils')(from, relation_alias, except, regex, prefix, suffix)) }}
 {% endmacro %}
 
-{% macro default__star(from, relation_alias=False, except=[], prefix='', suffix='') -%}
+{% macro default__star(from, relation_alias=False, except=[], regex='', prefix='', suffix='') -%}
```
Comment (dbeatty10) [inline macros/sql/star.sql:5]: I'd prefer to add the optional `regex` argument to the end of the signature. That way, this new feature will be guaranteed to be non-breaking for all users.

Do you have any thoughts about the order of parameters?

I think the change would be as simple as the following:

```sql
{% macro star(from, relation_alias=False, except=[], prefix='', suffix='', regex='') -%}
    {{ return(adapter.dispatch('star', 'dbt_utils')(from, relation_alias, except, prefix, suffix, regex)) }}
{% endmacro %}

{% macro default__star(from, relation_alias=False, except=[], prefix='', suffix='', regex='') -%}
```

**Review by dbeatty10 — CHANGES_REQUESTED** [review]
I'd prefer to add the optional regex argument to the end of the signature. That way, this new feature will be guaranteed to be non-breaking for all users.

Do you have any thoughts about the order of parameters?

See inline comments for suggested updates.

**Comment by robertdefilippi** [comment]
I'm going to close this PR and reopen with a fresh branch.

**Pull request closed without merging**